src/main/java/com/gmail/goosius/siegewar/events/PreSiegeWarStartEvent.java

```diff
@@ -0,0 +1,78 @@
+package com.gmail.goosius.siegewar.events;
+
+import com.gmail.goosius.siegewar.SiegeWar;
+import com.gmail.goosius.siegewar.settings.Translation;
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class PreSiegeWarStartEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private final Town townOfAttackingPlayer;
+    private final Nation nationOfAttackingPlayer;
+    private final Block flag;
+    private final TownBlock townBlock;
+    private final Town defendingTown;
+    private boolean isCancelled;
+    private String cancellationMsg = SiegeWar.prefix+"Siege prevented by another plugin.";
```

Comment:
The SiegeWar.prefix isn't required here, the prefix will be added by the try/catch which catches the TownyException.